analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Nullable.cs

```diff
@@ -118,4 +118,23 @@ public void Nullable_Ctor_Argument_PropagateConstraints()
         validator.ValidateTag("IsFalse", x => x.AllConstraints.Select(x => x.ToString()).OrderBy(x => x).JoinStr(", ").Should().Be("False, First, NotNull"));
         validator.ValidateTag("IsInt", x => x.AllConstraints.Select(x => x.ToString()).OrderBy(x => x).JoinStr(", ").Should().Be("NotNull"));
     }
+
+    [TestMethod]
+    public void Nullable_Conversion_PropagateConstraints()
+    {
+        const string code = """
+            var isTrue = true;
+            bool? toNullableImplicit = isTrue;
+            bool? toNullableExplicit = (bool?)isTrue;
+            bool toBoolExplicit = (bool)toNullableImplicit;
```

Comment:
A cast in an `as` expression is missing: `isTrue as bool?`

Reply:
NIT: There might be a user-defined conversion from bool to another struct (e.g. `Maybe`). Do we want to move the bool constraint there too? Might be worth adding a test case to document the current behavior.
```cs
Maybe m1 = true; // Does "m1" have BoolConstraint.True afterward?
Maybe? m2 = true; // If so, it does have it here too.
```
I created a sample on [sharplab](https://sharplab.io/#v2:EYLgtghglgdgNAFxAJwK7wCYgNQB8ACATAIwCwAUBQLIQCewApgPwAEYLAvCwmgwNwV8xAJwAKMAEoBlcgGceqAMYIWNegwoBvCi10tgAewMAbVpHXS9LfAGZVdRqMMmWAM2QHJnAHxsHDTjcPMD4rHT1ba2IANhYoMAAHYyhFKBUDBIZkCAQDZHt1JyNjfQkfFhgGAHcCx2ApcN1GlgB6FpYAKVR5Nzy442NunhyoAxgQZsjnEoBJWQAVXnLzRk4uBX5Ju2mWOYAxCGNZAI5fFZOuV0Pjywjt4t3ZAFUYAGsYAyqYZf81itQBtItiwDAA3LLIKAYAJCAAMLHmBgAyjxYABzURlU5+dRMAB0iJRkJgGLKTFYACIYADjBTpABfCgUIA==).

Reply:
The custom operator is a good catch. 
As that method could do anything, to a completely different type, we can't assume anything about the produced value (could be even from non-nullable to a null). I've updated the conversion implementation